addOns/paramminer/src/main/java/org/zaproxy/addon/paramminer/UrlUtils.java

```diff
@@ -0,0 +1,93 @@
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.addon.paramminer;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.commons.httpclient.URI;
+import org.apache.commons.httpclient.URIException;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.parosproxy.paros.model.Model;
+import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.network.HttpRequestHeader;
+import org.parosproxy.paros.network.HttpSender;
+import org.zaproxy.zap.network.HttpRequestConfig;
+
+public class UrlUtils {
+    private static Logger logger = LogManager.getLogger(UrlUtils.class);
+    private static HttpSender sender =
+            new HttpSender(
+                    Model.getSingleton().getOptionsParam().getConnectionParam(),
+                    true,
+                    HttpSender.MANUAL_REQUEST_INITIATOR);
+    private static HttpRequestConfig config = HttpRequestConfig.builder().setSoTimeout(10).build();
+
+    public static List<String> reader(String path) {
+        logger.info("Reading keys from file: " + path);
+        List<String> params = new ArrayList<>();
+        try {
+            BufferedReader br = new BufferedReader(new FileReader(path));
+            String line;
+            while ((line = br.readLine()) != null) {
+                params.add(StringUtils.strip(line));
+            }
+            br.close();
+            logger.info("Succesfully parsed keys!");
+        } catch (Exception e) {
+            logger.error(e);
+        }
+        return params;
+    }
+
+    public static HttpMessage stableRequest(String url, HttpRequestHeader headers) {
+        try {
+            URI parsed = new URI(url, true);
+            String scheme = parsed.getScheme();
+            String host = parsed.getHost();
+            String path = parsed.getPath();
+            String[] schemes =
+                    (scheme.equalsIgnoreCase("https"))
+                            ? new String[] {"https", "http"}
+                            : new String[] {"http", "https"};
+            for (String s : schemes) {
+                try {
+                    HttpMessage msg = new HttpMessage(new URI(s + "://" + host + path, true));
+                    msg.setRequestHeader(headers);
+                    sender.sendAndReceive(msg);
+                    return msg;
+                } catch (Exception e) {
+                    // Check if it is a connection error
+                    if (e.getMessage().contains("ConnectionError")
```

Comment:
Might be worth counting these and logging it after the loop

Reply:
Counting errors?

Reply:
Yup